### ISSUE TITLE
If runconfig doesn't exist, return None

### ISSUE THREAD

**Issue by skrauselevy** [TresAmigosSD/SMV#1252]
Currently if a runconfig doesn't exist we return an error, we should be returning None (Case 2).

1. In the case of running a module that is trying to access a key|value pair where the key is defined but there is no value ```smvGetRunConfig`` returns None and no error is thrown.

2.  In the case of running a module that is trying to access a key|value pair that does not exist the below error is thrown:

```
Traceback (most recent call last):
  File "/Users/skrauselevy/DataSenseAnalytics/SMV/tools/../src/main/python/scripts/runapp.py", line 19, in <module>
    app.run()
  File "/Users/skrauselevy/DataSenseAnalytics/SMV/src/main/python/smv/smvapp.py", line 434, in run
    self.j_smvApp.run()
  File "/Users/skrauselevy/DataSenseAnalytics/spark/python/lib/py4j-0.8.2.1-src.zip/py4j/java_gateway.py", line 538, in __call__
  File "/Users/skrauselevy/DataSenseAnalytics/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 36, in deco
  File "/Users/skrauselevy/DataSenseAnalytics/spark/python/lib/py4j-0.8.2.1-src.zip/py4j/protocol.py", line 300, in get_return_value
py4j.protocol.Py4JJavaError: An error occurred while calling o23.run.
: org.tresamigos.smv.SmvRuntimeException: There was an error executing Python code
	at org.tresamigos.smv.python.InterfacesWithPy4J$class.getPy4JResult(Py4JInterface.scala:28)
	at org.tresamigos.smv.SmvExtModulePython.getPy4JResult(SmvDataSet.scala:1124)
	at org.tresamigos.smv.SmvExtModulePython.doRun(SmvDataSet.scala:1143)
	at org.tresamigos.smv.SmvDataSet$$anonfun$computeRDD$1$$anonfun$applyOrElse$1$$anonfun$apply$1.applyOrElse(SmvDataSet.scala:534)
	at org.tresamigos.smv.SmvDataSet$$anonfun$computeRDD$1$$anonfun$applyOrElse$1$$anonfun$apply$1.applyOrElse(SmvDataSet.scala:533)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:33)
	at scala.util.Failure.recoverWith(Try.scala:172)
	at org.tresamigos.smv.SmvDataSet$$anonfun$computeRDD$1$$anonfun$applyOrElse$1.apply(SmvDataSet.scala:533)
	at org.tresamigos.smv.SmvDataSet$$anonfun$computeRDD$1$$anonfun$applyOrElse$1.apply(SmvDataSet.scala:533)
	at org.tresamigos.smv.SmvLock$.withLock(SmvLock.scala:70)
	at org.tresamigos.smv.SmvDataSet$$anonfun$computeRDD$1.applyOrElse(SmvDataSet.scala:530)
	at org.tresamigos.smv.SmvDataSet$$anonfun$computeRDD$1.applyOrElse(SmvDataSet.scala:528)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:33)
	at scala.util.Failure.recoverWith(Try.scala:172)
	at org.tresamigos.smv.SmvDataSet.computeRDD(SmvDataSet.scala:528)
	at org.tresamigos.smv.SmvDataSet.rdd(SmvDataSet.scala:222)
	at org.tresamigos.smv.SmvApp$$anonfun$generateOutputModules$1.apply(SmvApp.scala:315)
	at org.tresamigos.smv.SmvApp$$anonfun$generateOutputModules$1.apply(SmvApp.scala:315)
	at scala.collection.immutable.List.foreach(List.scala:318)
	at org.tresamigos.smv.SmvApp.generateOutputModules(SmvApp.scala:315)
	at org.tresamigos.smv.SmvApp.run(SmvApp.scala:484)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:231)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:379)
	at py4j.Gateway.invoke(Gateway.java:259)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:133)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:207)
	at java.lang.Thread.run(Thread.java:748)
Caused by: org.tresamigos.smv.python.SmvPythonException: The following error occurred whiling calling back to Python code
Traceback (most recent call last):
  File "/Users/skrauselevy/DataSenseAnalytics/SMV/src/main/python/smv/py4j_interface.py", line 35, in interface_method
    result = impl_method(*args)
  File "/Users/skrauselevy/DataSenseAnalytics/SMV/src/main/python/smv/smvdataset.py", line 693, in doRun
    result = self.run(i)
  File "/Users/skrauselevy/DataSenseAnalytics/Test_SMV/src/main/python/stage1/employment.py", line 20, in run
    two = self.smvGetRunConfig("ten")
  File "/Users/skrauselevy/DataSenseAnalytics/SMV/src/main/python/smv/runconfig.py", line 30, in smvGetRunConfig
    return SmvApp.getInstance().j_smvPyClient.getRunConfig(key)
  File "/Users/skrauselevy/DataSenseAnalytics/spark/python/lib/py4j-0.8.2.1-src.zip/py4j/java_gateway.py", line 538, in __call__
    self.target_id, self.name)
  File "/Users/skrauselevy/DataSenseAnalytics/spark/python/lib/pyspark.zip/pyspark/sql/utils.py", line 36, in deco
    return f(*a, **kw)
  File "/Users/skrauselevy/DataSenseAnalytics/spark/python/lib/py4j-0.8.2.1-src.zip/py4j/protocol.py", line 300, in get_return_value
    format(target_id, '.', name), value)
Py4JJavaError: An error occurred while calling o22.getRunConfig.
: java.util.NoSuchElementException: key not found: smv.config.ten
	at scala.collection.MapLike$class.default(MapLike.scala:228)
	at scala.collection.AbstractMap.default(Map.scala:58)
	at scala.collection.MapLike$class.apply(MapLike.scala:141)
	at scala.collection.AbstractMap.apply(Map.scala:58)
	at org.tresamigos.smv.SmvConfig$$anonfun$getRunConfig$1.apply(SmvConfig.scala:273)
	at org.tresamigos.smv.SmvConfig$$anonfun$getRunConfig$1.apply(SmvConfig.scala:273)
	at scala.collection.MapLike$class.getOrElse(MapLike.scala:128)
	at scala.collection.AbstractMap.getOrElse(Map.scala:58)
	at org.tresamigos.smv.SmvConfig.getRunConfig(SmvConfig.scala:273)
	at org.tresamigos.smv.python.SmvPyClient.getRunConfig(SmvPythonProxy.scala:360)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:231)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:379)
	at py4j.Gateway.invoke(Gateway.java:259)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:133)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:207)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by laneb** [comment]
This is duplicate of https://github.com/TresAmigosSD/SMV/issues/1176 which has some discussion that I think is valuable.

**Comment by laneb** [comment]
@skrauselevy Can we close this issue? I think it was resolved by #1253.

**Comment by cloudysail** [comment]
Aha, this one should have been automatically closed if `#1252` was added to the description of #1253
